### PR TITLE
add vae image processor

### PR DESCRIPTION
**Pull request by echarlaix**
In this PR we add an `VaeImageProcessor` class enabling the preprocessing and postprocessing of images pipelines, in which we correctly reshape the inputs as expected by the ONNX model. The watermark is also made optional.

### CODE REVIEW

File: .github/workflows/test_onnxruntime.yml
Comment (echarlaix) [inline .github/workflows/test_onnxruntime.yml:32]: waiting for `diffusers` next release before we can merge

**Review by fxmarty — APPROVED** [review]
LGTM

File: optimum/onnxruntime/modeling_diffusion.py
Comment (echarlaix) [reply optimum/onnxruntime/modeling_diffusion.py:609]: No without making `invisible-watermark` a hard dependency, I added an additional verification in https://github.com/huggingface/optimum/pull/1219/commits/dd292d8aa78518889ca99d54b76697141bb84e5b to check that `invisible-watermark` is installed if needed

**Pull request merged**
